cpp/tests/search/search_list_test.cpp

```diff
@@ -347,3 +347,66 @@ TYPED_TEST(TypedListContainsTestColumnNeedles, ListsOfStructs)
   auto const result   = cudf::contains(*haystack, *needles);
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, *result, verbosity);
 }
+
+struct ListLowerBound : public cudf::test::BaseFixture {
+};
+
+TEST_F(ListLowerBound, ListWithNulls)
+{
+  {
+    using lcw = cudf::test::lists_column_wrapper<double>;
+    using cudf::test::iterators::nulls_at;
+    lcw col1{
+      lcw{-3.45967821e+12},  // 0
+      lcw{-3.6912186e-32},   // 1
+      lcw{9.721175},         // 2
+    };
+
+    lcw val1{
+      lcw{{0, 4.22671e+32}, nulls_at({0})},
+    };
+
+    cudf::table_view input{{col1}};
+    cudf::table_view values{{val1}};
+    std::vector<cudf::order> column_order{cudf::order::ASCENDING};
+    std::vector<cudf::null_order> null_order_flags{cudf::null_order::BEFORE};
+
+    auto expect = cudf::test::fixed_width_column_wrapper<cudf::size_type>{0};
+    auto result = cudf::lower_bound(input, values, column_order, null_order_flags);
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expect, *result);
+  }
+
+  {
+    using lcw = cudf::test::lists_column_wrapper<int32_t, int32_t>;
+    using cudf::test::iterators::nulls_at;
+    lcw col1{
```

Comment:
This and similar below:
```suggestion
    auto const col1 = lcw{
```